ampl-converter/src/main/java/com/powsybl/ampl/converter/version/AmplExportVersion.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.ampl.converter.version;
+
+import com.powsybl.ampl.converter.AmplExportConfig;
+import com.powsybl.ampl.converter.AmplSubset;
+import com.powsybl.commons.util.StringToIntMapper;
+import com.powsybl.iidm.network.Network;
+
+/**
+ * @author Nicolas Pierre <nicolas.pierre at artelys.com>
+ */
+public interface AmplExportVersion {
```

Comment:
In Powsybl, I often see interfaces with only one implementation, is it considered a good practice on this project to systematically create an interface for each class?

Reply:
We generally use interfaces when we want to allow another project to use its own implementation instead of the one provided.
For instance, in `powsybl-core` we have the `Line` interface and an implementation [`LineImpl`](https://github.com/powsybl/powsybl-core/blob/main/iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/LineImpl.java) (you can note that they are not in the same package: respectively `com.powsybl.iidm.network` and `com.powsybl.iidm.network.impl`). But the `powsybl-network-store` project uses [its own implementation](https://github.com/powsybl/powsybl-network-store/blob/main/network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/LineImpl.java) which is quite different from the one provided by `powsybl-core`.

Here, I don't think we want to allow another project to define its own versions enum. Since `AmplExporter` uses `AmplExportVersionImpl.values()` [to retrieve](https://github.com/powsybl/powsybl-core/blob/50ad7ebd7b392e2c6892cc6359e6b9f38fbfa0a4/ampl-converter/src/main/java/com/powsybl/ampl/converter/AmplExporter.java#L100) the `AmplExportVersion` value to use, it should also be rewritten in order to use custom values anyway.

Reply:
Thank you for the explanation